### PR TITLE
Apply classes Lock, Unlock and add Script for multi-signature, flash layer

### PR DESCRIPTION
**Pull request by MichaelKim20**
1. Changed the structure of the Transaction
    Replace Signature of TxInput with Unlock, replace PublicKey of TxOutput with Lock
2. Changed signer of Transaction
3. Changed TxBuilder
4. Added a module Lock
5. Added a module Script, Opcodes

It was made for a flash-layer of Agora.
I think we should create a temporary branch similar to Agora.

Depend on #75 

Related to #74

### CODE REVIEW

**Comment by MichaelKim20** [comment]
I created a PR that uses it on Stoa so that I can test this PR.

https://github.com/bpfkorea/stoa/pull/248

**Comment by MichaelKim20** [comment]
I added a class `Script` to be used in Lock and Unlock

**Comment by MichaelKim20** [comment]
Rebase has been made to the timestamp of the block.

**Comment by MichaelKim20** [comment]
This PR is mostly translated code from D, so we will merge without review.

File: src/modules/script/Script.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+/**
+ * Contains a syntactically validated set of script opcodes
+ */
+export class Script
+{
+    /**
+     * opcodes and any associated data for each push opcode
+     */
+    private readonly _opcodes : Buffer;
```
Comment (TrustHenry) [inline src/modules/script/Script.ts:47]: `    private readonly _opcodes: Buffer;`

**Review by TrustHenry — COMMENTED** [review]
a few style fix.

**Pull request merged**